### ISSUE TITLE
Cannot run auto

### ISSUE THREAD

**Issue by AllenMuenLee** [purduesigbots/VOSS#109]
Here's the code:
`#include "main.h"
#include "bits/stdc++.h"
#include "VOSS/api.hpp"

auto ec = voss::controller::ExitConditions::new_conditions()
              .add_settle(400, 0.5, 400)
              .add_tolerance(1.0, 2.0, 200)
              .add_timeout(22500)
              .add_thru_smoothness(4)
              .build();

auto odom = voss::localizer::TrackingWheelLocalizerBuilder::new_builder()
                .with_right_motor(-9)
                .with_left_motor(2)
                .with_track_width(12.5)
                .with_left_right_tpi(6.911498)
                .with_imu(16)
                .build();

auto pid = voss::controller::PIDControllerBuilder::new_builder(odom)
               .with_linear_constants(20, 0.02, 169)
               .with_angular_constants(250, 0.05, 2435)
               .with_min_error(5)
               .with_min_vel_for_thru(100)
               .build();  
/**
 * A callback function for LLEMU's center button.
 *
 * When this callback is fired, it will toggle line 2 of the LCD text between
 * "I was pressed!" and nothing.
 */
void on_center_button() {
	
}

/**
 * Runs initialization code. This occurs as soon as the program is started.
 *
 * All other competition modes are blocked by initialize; it is recommended
 * to keep execution time for this mode under a few seconds.
 */
void initialize() {
	pros::lcd::initialize();
    odom->begin_localization(); //calibrate and begin localizing
    pros::screen::set_pen(pros::Color::red);
}

/**
 * Runs while the robot is in the disabled state of Field Management System or
 * the VEX Competition Switch, following either autonomous or opcontrol. When
 * the robot is enabled, this task will exit.
 */
void disabled() {}

/**
 * Runs after initialize(), and before autonomous when connected to the Field
 * Management System or the VEX Competition Switch. This is intended for
 * competition-specific initialization routines, such as an autonomous selector
 * on the LCD.
 *
 * This task will exit when the robot is enabled and autonomous or opcontrol
 * starts.
 */
void competition_initialize() {}

/**
 * Runs the user autonomous code. This function will be started in its own task
 * with the default priority and stack size whenever the robot is enabled via
 * the Field Management System or the VEX Competition Switch in the autonomous
 * mode. Alternatively, this function may be called in initialize or opcontrol
 * for non-competition testing purposes.
 *
 * If the robot is disabled or communications is lost, the autonomous task
 * will be stopped. Re-enabling the robot will restart the task, not re-start it
 * from where it left off.
 */

/**
 * Runs the operator control code. This function will be started in its own task
 * with the default priority and stack size whenever the robot is enabled via
 * the Field Management System or the VEX Competition Switch in the operator
 * control mode.
 *
 * If no competition control is connected, this function will run immediately
 * following initialize().
 *
 * If the robot is disabled or communications is lost, the
 * operator control task will be stopped. Re-enabling the robot will restart the
 * task, not resume it from where it left off.
 */

#define LEFT_MOTORS {-1, 2, -3}
#define RIGHT_MOTORS {8, -9, 10}

pros::Motor Intake(7, pros::v5::MotorGear::blue);
pros::Motor wall(6, pros::v5::MotorGear::red);
pros::ADIDigitalOut piston ('a');

bool piston_bool = false;

auto chassis = voss::chassis::DiffChassis(LEFT_MOTORS, RIGHT_MOTORS, pid, ec, 8, pros::E_MOTOR_BRAKE_BRAKE);
pros::Controller master(pros::E_CONTROLLER_MASTER);
//we recommend using the pid controller as default controller

void intake_contr(){
    if (master.get_digital(DIGITAL_L1)){
        //pros::screen::print(TEXT_MEDIUM, 1, "L1");
        Intake.move_voltage(12000);
    } else if (master.get_digital(DIGITAL_L2)){
        //pros::screen::print(TEXT_MEDIUM, 1, "L2");
        Intake.move_voltage(-12000);
    } else {
        Intake.move_velocity(0);
    }
}

void wall_contr(){
    int count = 0;
    char c = (char)wall.get_position();
    if (master.get_digital(DIGITAL_UP)){
        wall.move_voltage(8000);
    } else if (master.get_digital(DIGITAL_DOWN)){
        //pros::screen::print(TEXT_MEDIUM, 1, "L2");
        wall.move_voltage(-5000);
    } else {
        wall.set_brake_mode(MOTOR_BRAKE_HOLD);
        wall.brake();
    }
}

void fixer(){
    if (master.get_digital_new_press(DIGITAL_A)){
        piston_bool = !piston_bool;
        pros::delay(200);
    }
}

void funcs(){
    intake_contr();
    fixer();
    wall_contr();

    piston.set_value(!piston_bool);
}

void autonomous() {
    Intake.move_voltage(12000);
    chassis.move(10);
}

void opcontrol() {
    wall.tare_position();
	while (true) {
        funcs();
		chassis.arcade(master.get_analog(ANALOG_LEFT_Y) * 0.6, master.get_analog(ANALOG_RIGHT_X) * 0.6);
		pros::delay(20);
	}
}`
Nothing happened when I run autonomous skill using controller.

**Comment by Rocky14683** [comment]
problem solved.
